### PR TITLE
GG-23732 DATE column changes value depending on time zone of JDBC client

### PR DESCRIPTION
**Pull request by mshonichev**

### CODE REVIEW

File: modules/clients/pom.xml [hunk 1]
```diff
@@ -132,6 +132,13 @@
             <version>${commons.lang3.version}</version>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
```
Comment (ygerzhedovich) [inline modules/clients/pom.xml]: Wow! what reason to add the dependency?

File: modules/clients/src/test/java/org/apache/ignite/qa/query/JdbcThinTimezoneTest.java [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+            assertFalse(res.isEmpty());
+
+            resMap.forEach((key, value) -> {
+                if (!value.equals(res)) {
+                    for (String s : res)
+                        System.err.println(tz + ", " + s);
+
+                    for (String s : value)
+                        System.err.println(key + ", " + s);
+
```
Comment (ygerzhedovich) [inline modules/clients/src/test/java/org/apache/ignite/qa/query/JdbcThinTimezoneTest.java]: should we remove the debug code?

File: modules/h2/src/main/java/org/h2/util/DateTimeUtils.java [hunk 3]
```diff
@@ -1582,4 +1582,11 @@ public static long convertScale(long nanosOfDay, int scale) {
         return nanosOfDay - mod;
     }
 
+    private static TimeZone getTimeZone() {
```
Comment (AMashenkov) [inline modules/h2/src/main/java/org/h2/util/DateTimeUtils.java]: Javadoc

File: modules/h2/src/main/java/org/h2/util/DateTimeUtils.java [hunk 4]
```diff
@@ -1582,4 +1582,11 @@ public static long convertScale(long nanosOfDay, int scale) {
...
+        return timeZone;
+    }
+
+    private static void setTimeZone(TimeZone tz) {
```
Comment (AMashenkov) [inline modules/h2/src/main/java/org/h2/util/DateTimeUtils.java]: Javadoc

File: modules/clients/src/test/java/org/apache/ignite/qa/query/JdbcThinTimezoneTest.java [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.testframework.TestJavaProcess;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+
+import static org.apache.ignite.jdbc.JdbcTestUtils.sql;
+
+/**
+ * Thin client authorization with Native Ignite authentication tests.
```
Comment (korlov42) [inline modules/clients/src/test/java/org/apache/ignite/qa/query/JdbcThinTimezoneTest.java]: misleading comment

File: modules/clients/src/test/java/org/apache/ignite/qa/query/JdbcThinTimezoneTest.java [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+    }
+
+    /** */
+    @Test
+    public void test() throws Exception {
```
Comment (korlov42) [inline modules/clients/src/test/java/org/apache/ignite/qa/query/JdbcThinTimezoneTest.java]: please name this test correctly

File: modules/clients/src/test/java/org/apache/ignite/jdbc/JdbcTestUtils.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Tests for complex queries (joins, etc.).
```
Comment (korlov42) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/JdbcTestUtils.java]: misleading comment

File: modules/clients/pom.xml [hunk 8]
```diff
@@ -132,6 +132,13 @@
...
+            <groupId>postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>9.1-901.jdbc4</version>
```
Comment (korlov42) [inline modules/clients/pom.xml]: but for what sake?

File: modules/clients/src/test/java/org/apache/ignite/qa/query/JdbcThinTimezoneTest.java (hunk 2 above)
Comment (tledkov-gridgain) [reply modules/clients/src/test/java/org/apache/ignite/qa/query/JdbcThinTimezoneTest.java]: I guess this is not debug print. This is the extended assertion info. I think it doesn't make sense. 
We print results that doesn't pass the check.
The test doesn't print any results when all passes.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/jdbc/JdbcUtils.java [hunk 9]
```diff
@@ -141,4 +155,274 @@ public static void writeNullableLong(BinaryWriterExImpl writer, @Nullable Long v
...
+     * @param protoCtx Protocol context.
+     * @return Read object.
+     * @throws BinaryObjectException On error.
+     */
+    @Nullable public static Object readObject(BinaryReaderExImpl reader, boolean binObjAllow,
+        JdbcProtocolContext protoCtx)
+        throws BinaryObjectException {
+        byte type = reader.readByte();
+
+        switch (type) {
```
[the remark this answers is not in the document]
Comment (tledkov-gridgain) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/jdbc/JdbcUtils.java]: Great solution. Fixed.

**Pull request closed without merging**